modules/layers/src/icon-layer/icon-layer.js

```diff
@@ -73,8 +74,20 @@ export default class IconLayer extends Layer {
   }
 
   initializeState() {
-    const attributeManager = this.getAttributeManager();
+    // if user provided iconAtlas, do not need `IconManager`
+    if (!this.props.iconAtlas) {
+      const {data, getIcon} = this.props;
+      const {gl} = this.context;
+      const iconManager = new IconManager(gl, {
```

Comment:
Creating the icon manager in here means that the user cannot change the usage mode later. If this is by design, or a temporary limitation, it needs to be documented.